src/main/java/org/candlepin/resource/GuestIdResource.java

```diff
@@ -141,10 +145,13 @@ public void deleteGuest(@Verify(Consumer.class) String consumerUuid, String gues
 
     @Override
     @RootResource.LinkedResource
-    public CandlepinQuery<GuestIdDTOArrayElement> getGuestIds(@Verify(Consumer.class) String consumerUuid) {
+    public Stream<GuestIdDTOArrayElement> getGuestIds(@Verify(Consumer.class) String consumerUuid) {
         Consumer consumer = consumerCurator.findByUuid(consumerUuid);
-        return translator.translateQuery(guestIdCurator.listByConsumer(consumer),
-            GuestIdDTOArrayElement.class);
+        List<GuestId> guestIds = guestIdCurator.listByConsumer(consumer);
+        Stream<GuestIdDTOArrayElement> stream = guestIds == null ? Stream.empty() : guestIds.stream()
+            .map(this.translator.getStreamMapper(GuestId.class, GuestIdDTOArrayElement.class));
+        return this.pagingUtilFactory.forClass(GuestIdDTOArrayElement.class)
+            .applyPaging(stream, guestIds.size());
```

Comment:
Something from Sonarqube that actually checks out: guestIds is implied to be nullable here, but we use it on line 154 as if it's not.

Since we should treat other classes as black boxes and we don't have any contract established around the nullability of the output of our curator methods, we should gate the whole thing around whether or not it's null:

```
List<GuestId> guestIds = guestIdCurator.listByConsumer(consumer);

if (guestIds != null) {
  Stream<GuestIdDTOArrayElement> stream = guestIds.stream()
    .map(this.translator.getStreamMapper(...);

  return this.pagingUtilFactory.forClass(...)
    .applyPaging(...);
}

return Stream.empty();
```